typewise_alert.py

```diff
@@ -1,46 +1,75 @@
+    if value < lower_limit:
+        return 'TOO_LOW'
+    if value > upper_limit:
+        return 'TOO_HIGH'
+    return 'NORMAL'
 
-def infer_breach(value, lowerLimit, upperLimit):
-  if value < lowerLimit:
-    return 'TOO_LOW'
-  if value > upperLimit:
-    return 'TOO_HIGH'
-  return 'NORMAL'
-
-
-def classify_temperature_breach(coolingType, temperatureInC):
-  lowerLimit = 0
-  upperLimit = 0
-  if coolingType == 'PASSIVE_COOLING':
-    lowerLimit = 0
-    upperLimit = 35
-  elif coolingType == 'HI_ACTIVE_COOLING':
-    lowerLimit = 0
-    upperLimit = 45
-  elif coolingType == 'MED_ACTIVE_COOLING':
-    lowerLimit = 0
-    upperLimit = 40
-  return infer_breach(temperatureInC, lowerLimit, upperLimit)
-
-
-def check_and_alert(alertTarget, batteryChar, temperatureInC):
-  breachType =\
-    classify_temperature_breach(batteryChar['coolingType'], temperatureInC)
-  if alertTarget == 'TO_CONTROLLER':
-    send_to_controller(breachType)
-  elif alertTarget == 'TO_EMAIL':
-    send_to_email(breachType)
-
-
-def send_to_controller(breachType):
-  header = 0xfeed
-  print(f'{header}, {breachType}')
-
-
-def send_to_email(breachType):
-  recepient = "a.b@c.com"
-  if breachType == 'TOO_LOW':
-    print(f'To: {recepient}')
-    print('Hi, the temperature is too low')
-  elif breachType == 'TOO_HIGH':
-    print(f'To: {recepient}')
-    print('Hi, the temperature is too high')
+
+class PassiveCooling:
+    lower_limit = 0
+    upper_limit = 35
+
+
+class HiActiveCooling:
+    lower_limit = 0
+    upper_limit = 45
+
+
+class MidActiveCooling:
+    lower_limit = 0
+    upper_limit = 40
+
+
+class CoolingTypeFactory:
+    def __init__(self):
+        self.temp_limit = {"PASSIVE_COOLING": PassiveCooling(),
+                           "HI_ACTIVE_COOLING": HiActiveCooling(),
+                           "MED_ACTIVE_COOLING": MidActiveCooling()}
+
+    def get_cooling_class(self, cooling_type):
+        return self.temp_limit.get(cooling_type)
+
+
+def get_temperature_limits(cooling_type):
+    cooling_type_obj = CoolingTypeFactory().get_cooling_class(cooling_type)
+    return cooling_type_obj.lower_limit, cooling_type_obj.upper_limit
+
+
+def classify_temperature_breach(cooling_type, temperature_in_c):
+    lower_limit, upper_limit = get_temperature_limits(cooling_type)
+    return infer_breach(temperature_in_c, lower_limit, upper_limit)
+
+
+def monitor_battery_temp(alert_target, battery_char, temperature_in_c):
+    breach_type = classify_temperature_breach(battery_char['coolingType'], temperature_in_c)
+    if alert_target == 'TO_CONTROLLER':
+        recipient, msg = create_alert_for_controller(breach_type)
+    elif alert_target == 'TO_EMAIL':
+        recipient, msg = create_alert_for_email(breach_type)
+    return send_alert(recipient, msg)
+
+
+def create_alert_for_controller(breach_type):
+    recipient = 0xfeed
+    msg = breach_type
+    return recipient, msg
+
+
+def create_alert_for_email(breach_type):
```

Comment:
This is a stub and the implementation has a strong dependency on it. Use dependency injection or mocking to flexibly switch between the stub and production code.

Reply:
@sudeeprp thanks for the review. I have tried to inject stubs to avoid this dependency while testing with a new commit. Would appreciate if you could take a look some time. Thank you